apps/cowswap-frontend/src/modules/bridge/hooks/useTryFindIntermediateToken.ts

```diff
@@ -1,24 +1,24 @@
 import { useMemo } from 'react'
 
 import { TokenWithLogo } from '@cowprotocol/common-const'
-import { BridgeQuoteResults } from '@cowprotocol/cow-sdk'
-import { useSearchToken, useTokensByAddressMap } from '@cowprotocol/tokens'
+import { BridgeQuoteResults, OrderKind } from '@cowprotocol/cow-sdk'
+import { useSearchToken } from '@cowprotocol/tokens'
+
+import { useGetIntermediateSellTokenFromOrder } from 'modules/trade/hooks/useGetIntermediateSellTokenFromOrder'
 
 interface UseGetMaybeIntermediateTokenProps {
   bridgeQuote: BridgeQuoteResults | null
 }
 
 // returns the intermediate buy token (if it exists) and if it should be imported
-export function useGetMaybeIntermediateToken({ bridgeQuote }: UseGetMaybeIntermediateTokenProps): {
+export function useTryFindIntermediateToken({ bridgeQuote }: UseGetMaybeIntermediateTokenProps): {
   intermediateBuyToken: TokenWithLogo | null
   toBeImported: boolean
 } {
   const { sellTokenAddress: intermediateBuyTokenAddress } = bridgeQuote?.tradeParameters || {}
-  const tokensByAddress = useTokensByAddressMap()
-
-  const intermediateBuyToken =
-    (intermediateBuyTokenAddress && tokensByAddress[intermediateBuyTokenAddress.toLowerCase()]) || null
+  const orderParams = getOrderParamsFromQuote(bridgeQuote)
 
+  const intermediateBuyToken = useGetIntermediateSellTokenFromOrder(orderParams) ?? null
```

Comment:
`orderParams` is passed into the hook and used in `useMemo` deps. But `getOrderParamsFromQuote` returns a new object every time. It need to be wrapped into useMemo()